unity/Assets/Scripts/Util/DataStructures/BST/AATree.cs

```diff
@@ -557,6 +575,25 @@ private void FindNodes(T data, Node t, List<T> list)
             }
         }
 
+        public IEnumerable<T> Iterator()
+        {
+            var result = new List<T>();
+            IterateNodes(m_Root, result);
+            result.Sort();
+            return result;
+        }
+
+        private void IterateNodes(Node t, List<T> list)
+        {
+            // return if at bottom node
+            if (t == m_Bottom) return;
+
+            list.Add(t.Data);
+
+            IterateNodes(t.Left, list);
+            IterateNodes(t.Right, list);
+        }
```

Comment:
Maybe change the return type to List and return the notes

Reply:
I think that will make it a lot slower in this case because now we are only instantiating a new `List` once, but otherwise we would need to create a new list for every invocation.